### PR TITLE
Show FormDiff modeless.

### PR DESCRIPTION
**Pull request by jbialobr**
What did I do to test the code and ensure quality:
 - Manual tests. 

Has been tested on (remove any that don't apply):
 - GIT 2.15
 - Windows 10

### CODE REVIEW

File: GitUI/UserControls/RevisionGrid.cs [hunk 1]
```diff
@@ -3732,11 +3732,10 @@ private void CompareToBranchToolStripMenuItem_Click(object sender, EventArgs e)
                 if (form.ShowDialog(this) == DialogResult.OK)
                 {
                     var baseCommit = Module.RevParse(form.BranchName);
-                    using (var diffForm = new FormDiff(UICommands, this, baseCommit, headCommit.Guid,
-                        form.BranchName, headCommit.Subject))
-                    {
-                        diffForm.ShowDialog(this);
-                    }
+                    var diffForm = new FormDiff(UICommands, this, baseCommit, headCommit.Guid,
```
Comment (pmiossec) [inline GitUI/UserControls/RevisionGrid.cs]: I think we could introduce a method, no!?!

**Comment by RussKie** [comment]
Could you please provide a bit more info on the nature of the change and the benefits it brings?

**Comment by drewnoakes** [comment]
Can't speak for @jbialobr but I am in favour of anything that converts modal forms to non-modal ones. Workflows with GE involve multiple windows and when you can't click between them it's really frustrating. I understand there are cases with technical reasons behind them, but they're just limitations of the design rather than some fundamental problem, so we should work towards removing modality where possible.

**Comment by jbialobr** [comment]
As @drewnoakes said, there are cases when you need some data from other forms (mostly main form) and when the FormDiff is modal you can't go to the main form to see the data. Scenario: you are in the middle of reviewing in FormDiff and want to read the commit message of the commit that introduced a change you are looking at.

I changed it to WIP because FormDiff depends on revisionGrid. I have to remove that dependency to make it work properly.

**Comment by RussKie** [comment]
Thank you.

**Pull request merged**